src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -13,35 +13,54 @@
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the task identified by the index number used in the last task listing.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
+            + ": Deletes the task identified by its name or the index number used in the last task listing.\n"
+            + "Parameters: INDEX (must be a positive integer) or NAME (the name of a task to delete)\n"
             + "Example: " + COMMAND_WORD + " 1";
 
     public static final String MESSAGE_DELETE_TASK_SUCCESS = "Deleted Task: %1$s";
 
     public final int targetIndex;
+    public final String targetName;
 
     public DeleteCommand(int targetIndex) {
         this.targetIndex = targetIndex;
+        this.targetName = null;
+    }
+
+    public DeleteCommand(String name) {
+        this.targetIndex = -1;
```

Comment:
Error:(32, 5) java: variable targetIndex might not have been initialized

when removed